src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp

```diff
@@ -463,10 +495,24 @@ ov::pass::RoPEFusionChatGLM::RoPEFusionChatGLM(int split_output_id) {
     auto var_split_2 = makePattern<opset1::VariadicSplit>({cos_sin_cache, 0, {0, ov::gen_pattern::Symbol("end")}});
     var_split_2->set_output_size(2);
 
-    auto view_Reshape_460 =
-        makePattern<opset1::Reshape>({slice_StridedSlice_449 | slice_Slice_449 | var_split_2->output(0),
-                                      ListConstruct_379_Concat | const_target_shape_2},
-                                     {{"special_zero", false}});
+    std::shared_ptr<ov::Node> view_Reshape_460 = nullptr;
+    if (support_2d_rope) {
+        // Slice cos_sin_cache to support 2-dimentional RoPE
+        auto ScatterUpdate = makePattern<opset3::ScatterUpdate>({{0, 0}, {1}, seq_length, {0}}, {});
+        auto slice_Slice_449 = makePattern<ov::opset8::Slice>({cos_sin_cache, {0,0}, ScatterUpdate, {1,1}, {0}});
+        auto slice_StridedSlice_449 = GenStridedSlice(cos_sin_cache, {0,0}, ScatterUpdate, {1,1}, 1);
+        // [batch, 1, seq_length, half_rotary_dims, 2]
+        view_Reshape_460 =
+            makePattern<opset1::Reshape>({slice_StridedSlice_449 | slice_Slice_449 | var_split_2->output(0),
+                                        ListConstruct_379_Concat | const_target_shape_2},
```

Comment:
The `const_target_shape_2` with `makeConst({batch, 1, seq_len, ndims / 2, 2});` could be replaced with `Concat`
```
auto cos_sin_shape = makeOP<opset1::Concat>({{-1}, {1}, seq_len, {ndims / 2}, {2}}, {{"axis", 0}});
```

Reply:
I think ListConstruct_379_Concat is cos_sin_shape, could you please check it again? thanks.

Reply:
Just find that the `ListConstruct_379_Concat` and `cos_sin_shape` are different, `ListConstruct_379_Concat` accepts all constant inputs, while `cos_sin_shape` accepts the `seq_len` not `seq_length`, which means that `seq_len` is a runtime value. Therefore `ListConstruct_379_Concat` cannot match `cos_sin_shape`